src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,161 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
```

Comment:
The object fields `head`, `tail` and `size` are correctly declared as private and are accessed through special methods. This is a good practice to encapsulate the data and maintain the integrity of the object.